### PR TITLE
add support of Qemu's user network

### PR DESCRIPTION
**Pull request by stlankes**
By setting the environment variable RUN_USER_PORT, the VM will use Qemu's user network and listen on the port RUN_USER_PORT.

### CODE REVIEW

File: src/hermit.rs [hunk 1]
```diff
@@ -37,10 +37,16 @@ pub fn prepare_environment(project_dir: &Path, hermit_env_path: &Option<PathBuf>
 	}
 }
 
+pub enum NetworkConfig {
+	VirtIO(network::VirtioNetworkConfig),
+	UserNetwork(u16),
+	None,
+}
```
Comment (mkroening) [inline src/hermit.rs:44]: I think this should either represent the Network backend (`tap|bridge|user|l2tpv3|vde|netmap|vhost-user|socket`) or the emulated network device (`rtl8139`, `e1000`, `virtio-net-pci`, etc.), but not mix those two dimensions.

https://www.qemu.org/docs/master/system/invocation.html#hxtool-5

**Review by mkroening — CHANGES_REQUESTED** [review]

File: src/hermit.rs (hunk 1 above)
Comment (stlankes) [reply src/hermit.rs]: I agree and fixed it.

**Review by mkroening — APPROVED** [review]
Looks good to me 👍

**Comment by stlankes** [comment]
bors +r

**Pull request merged**